### PR TITLE
Added relationship element and operation types

### PR DESCRIPTION
**Pull request by Taletex**

### CODE REVIEW

File: models/coreaas.xml [hunk 1]
```diff
@@ -2498,6 +2499,52 @@
...
+        </References>
+    </UAVariable>
+    <UAVariable DataType="String" ParentNodeId="ns=1;i=1021" NodeId="ns=1;i=6093" BrowseName="1:terminalType" AccessLevel="3">
+        <DisplayName>terminalType</DisplayName>
+        <References>
+            <Reference ReferenceType="HasModellingRule">i=78</Reference>
+            <Reference ReferenceType="HasTypeDefinition">i=68</Reference>
+            <Reference ReferenceType="HasProperty" IsForward="false">ns=1;i=1021</Reference>
+        </References>
+    </UAVariable>
```
Comment (msalafia) [inline models/coreaas.xml:2547]: Terminal stuff is still here!

File: models/coreaas.xml [hunk 2]
```diff
@@ -2766,6 +2813,62 @@ AAS or a reference to an external object or entity.</Description>
...
     </UAVariable>
+    <UAObjectType NodeId="ns=1;i=1019" BrowseName="1:OperationType">
+        <DisplayName>OperationType</DisplayName>
+        <References>
+            <Reference ReferenceType="HasComponent">ns=1;i=7001</Reference>
+            <Reference ReferenceType="HasSubtype" IsForward="false">ns=1;i=1010</Reference>
+        </References>
+    </UAObjectType>
+    <UAMethod ParentNodeId="ns=1;i=1019" NodeId="ns=1;i=7001" BrowseName="1:act">
+        <DisplayName>act</DisplayName>
```
Comment (msalafia) [inline models/coreaas.xml:2824]: Change it with Run or Execute.

File: models/coreaas.xml [hunk 3]
```diff
@@ -2766,6 +2813,62 @@ AAS or a reference to an external object or entity.</Description>
...
+        </References>
+    </UAObject>
+    <UAVariable DataType="Key" ParentNodeId="ns=1;i=5072" ValueRank="1" NodeId="ns=1;i=6091" ArrayDimensions="0" BrowseName="1:keys" AccessLevel="3">
+        <DisplayName>keys</DisplayName>
+        <References>
+            <Reference ReferenceType="HasProperty" IsForward="false">ns=1;i=5072</Reference>
+            <Reference ReferenceType="HasModellingRule">i=78</Reference>
+            <Reference ReferenceType="HasTypeDefinition">i=68</Reference>
+        </References>
+    </UAVariable>
```
Comment (msalafia) [inline models/coreaas.xml:2854]: Is this necessary? Shouldn't it inherit keys from its ObjectType?

File: models/coreaas.xml [hunk 4]
```diff
@@ -2766,6 +2813,62 @@ AAS or a reference to an external object or entity.</Description>
...
+        </References>
+    </UAObject>
+    <UAVariable DataType="Key" ParentNodeId="ns=1;i=5073" ValueRank="1" NodeId="ns=1;i=6092" ArrayDimensions="0" BrowseName="1:keys" AccessLevel="3">
+        <DisplayName>keys</DisplayName>
+        <References>
+            <Reference ReferenceType="HasProperty" IsForward="false">ns=1;i=5073</Reference>
+            <Reference ReferenceType="HasModellingRule">i=78</Reference>
+            <Reference ReferenceType="HasTypeDefinition">i=68</Reference>
+        </References>
+    </UAVariable>
```
Comment (msalafia) [inline models/coreaas.xml:2871]: Same question as above

**Comment by Taletex** [comment]
Marco, non so perché, se per sbaglio o automaticamente, ma nonostante io avessi solo fatto un push nella mia repository, ha fatto pure la pull request! Dopo, avendo visto che l'xml era ancora li, abbiamo deciso di forkare nuovamente la repo e di farlo pulito da 0. Tra poco ti inoltro anche l'xml corretto, che siamo finalmente riusciti a generare con il progetto in configurazione modelling.
________________________________
Da: Marco Salafia <notifications@github.com>
Inviato: giovedì 14 novembre 2019 11:50
A: OPCUAUniCT/coreAAS <coreAAS@noreply.github.com>
Cc: Alessandro <alessandromessina95@hotmail.com>; Author <author@noreply.github.com>
Oggetto: Re: [OPCUAUniCT/coreAAS] Added relationship element and operation types (#2)


@msalafia commented on this pull request.

________________________________

In models/coreaas.xml<https://github.com/OPCUAUniCT/coreAAS/pull/2#discussion_r346242452>:

> +        <References>
+            <Reference ReferenceType="HasSubtype" IsForward="false">ns=1;i=1005</Reference>
+            <Reference ReferenceType="HasProperty">ns=1;i=6095</Reference>
+            <Reference ReferenceType="HasProperty">ns=1;i=6094</Reference>
+            <Reference ReferenceType="HasProperty">ns=1;i=6093</Reference>
+        </References>
+    </UAObjectType>
+    <UAVariable DataType="Identifier" ParentNodeId="ns=1;i=1021" NodeId="ns=1;i=6095" BrowseName="1:identification" AccessLevel="3">
+        <DisplayName>identification</DisplayName>
+        <References>
+            <Reference ReferenceType="HasProperty" IsForward="false">ns=1;i=1021</Reference>
+            <Reference ReferenceType="HasModellingRule">i=78</Reference>
+            <Reference ReferenceType="HasTypeDefinition">i=68</Reference>
+        </References>
+        <Value>
+            <uax:ExtensionObject>
+                <uax:TypeId>
+                    <uax:Identifier>ns=1;i=5002</uax:Identifier>
+                </uax:TypeId>
+                <uax:Body>
+                    <Identifier xmlns="http://dieei.unict.it/coreAAS/Types.xsd">
+                        <id>http://www.test.com/terminalDataTemplate</id>
+                        <idType>URI_1</idType>
+                    </Identifier>
+                </uax:Body>
+            </uax:ExtensionObject>
+        </Value>
+    </UAVariable>
+    <UAVariable DataType="Int16" ParentNodeId="ns=1;i=1021" NodeId="ns=1;i=6094" BrowseName="1:terminalNumber" AccessLevel="3">
+        <DisplayName>terminalNumber</DisplayName>
+        <References>
+            <Reference ReferenceType="HasModellingRule">i=78</Reference>
+            <Reference ReferenceType="HasTypeDefinition">i=68</Reference>
+            <Reference ReferenceType="HasProperty" IsForward="false">ns=1;i=1021</Reference>
+        </References>
+    </UAVariable>
+    <UAVariable DataType="String" ParentNodeId="ns=1;i=1021" NodeId="ns=1;i=6093" BrowseName="1:terminalType" AccessLevel="3">
+        <DisplayName>terminalType</DisplayName>
+        <References>
+            <Reference ReferenceType="HasModellingRule">i=78</Reference>
+            <Reference ReferenceType="HasTypeDefinition">i=68</Reference>
+            <Reference ReferenceType="HasProperty" IsForward="false">ns=1;i=1021</Reference>
+        </References>
+    </UAVariable>


Terminal stuff is still here!

________________________________

In models/coreaas.xml<https://github.com/OPCUAUniCT/coreAAS/pull/2#discussion_r346243202>:

> @@ -2766,6 +2813,62 @@ AAS or a reference to an external object or entity.</Description>
             <Reference ReferenceType="HasProperty" IsForward="false">ns=1;i=1012</Reference>
         </References>
     </UAVariable>
+    <UAObjectType NodeId="ns=1;i=1019" BrowseName="1:OperationType">
+        <DisplayName>OperationType</DisplayName>
+        <References>
+            <Reference ReferenceType="HasComponent">ns=1;i=7001</Reference>
+            <Reference ReferenceType="HasSubtype" IsForward="false">ns=1;i=1010</Reference>
+        </References>
+    </UAObjectType>
+    <UAMethod ParentNodeId="ns=1;i=1019" NodeId="ns=1;i=7001" BrowseName="1:act">
+        <DisplayName>act</DisplayName>


Change it with Run or Execute.

________________________________

In models/coreaas.xml<https://github.com/OPCUAUniCT/coreAAS/pull/2#discussion_r346244306>:

> +    <UAVariable DataType="Key" ParentNodeId="ns=1;i=5072" ValueRank="1" NodeId="ns=1;i=6091" ArrayDimensions="0" BrowseName="1:keys" AccessLevel="3">
+        <DisplayName>keys</DisplayName>
+        <References>
+            <Reference ReferenceType="HasProperty" IsForward="false">ns=1;i=5072</Reference>
+            <Reference ReferenceType="HasModellingRule">i=78</Reference>
+            <Reference ReferenceType="HasTypeDefinition">i=68</Reference>
+        </References>
+    </UAVariable>


Is this necessary? Shouldn't it inherit keys from its ObjectType?

________________________________

In models/coreaas.xml<https://github.com/OPCUAUniCT/coreAAS/pull/2#discussion_r346244467>:

> +    <UAVariable DataType="Key" ParentNodeId="ns=1;i=5073" ValueRank="1" NodeId="ns=1;i=6092" ArrayDimensions="0" BrowseName="1:keys" AccessLevel="3">
+        <DisplayName>keys</DisplayName>
+        <References>
+            <Reference ReferenceType="HasProperty" IsForward="false">ns=1;i=5073</Reference>
+            <Reference ReferenceType="HasModellingRule">i=78</Reference>
+            <Reference ReferenceType="HasTypeDefinition">i=68</Reference>
+        </References>
+    </UAVariable>


Same question as above

**Pull request closed without merging**